internal/gatewayapi/contexts.go

```diff
@@ -73,6 +75,9 @@ type ListenerContext struct {
 	listenerStatusIdx int
 	namespaceSelector labels.Selector
 	tlsSecrets        []*corev1.Secret
+	tlsCertificates   []*x509.Certificate
```

Comment:
this is going to take up a lot of mem, can we instead only save DNSNames ?

Reply:
although its a pointer here, its avoiding gc of the underlying memory

Reply:
Yeah, good catch! Raw cert data can be huge.